client/db/bolt/db.go

```diff
@@ -532,24 +552,30 @@ func (db *BoltDB) marketOrdersSince(dexB, baseB, quoteB []byte, n int, since uin
 		timeB := oBkt.Get(updateTimeKey)
 		return bEqual(dexB, oBkt.Get(dexKey)) && bEqual(baseB, oBkt.Get(baseKey)) &&
 			bEqual(quoteB, oBkt.Get(quoteKey)) && bytes.Compare(timeB, sinceB) >= 0
-	})
+	}, true)
 }
 
 // newestOrders returns the n newest orders, filtered with a supplied filter
 // function. Each order's bucket is provided to the filter, and a boolean true
 // return value indicates the order should is eligible to be decoded and
 // returned.
-func (db *BoltDB) newestOrders(n int, filter func([]byte, *bbolt.Bucket) bool) ([]*dexdb.MetaOrder, error) {
+func (db *BoltDB) newestOrders(n int, filter func([]byte, *bbolt.Bucket) bool, allOrders bool) ([]*dexdb.MetaOrder, error) {
 	var orders []*dexdb.MetaOrder
-	return orders, db.ordersView(func(master *bbolt.Bucket) error {
-		pairs := newestBuckets(master, n, updateTimeKey, filter)
-		for _, pair := range pairs {
-			oBkt := master.Bucket(pair.k)
-			o, err := decodeOrderBucket(pair.k, oBkt)
-			if err != nil {
-				return err
+	return orders, db.ordersView(func(ob, aob *bbolt.Bucket) error {
+		buckets := []*bbolt.Bucket{ob}
+		if allOrders {
+			buckets = append(buckets, aob)
+		}
+		for _, master := range buckets {
+			pairs := newestBuckets(master, n, updateTimeKey, filter)
+			for _, pair := range pairs {
+				oBkt := master.Bucket(pair.k)
+				o, err := decodeOrderBucket(pair.k, oBkt)
+				if err != nil {
+					return err
+				}
+				orders = append(orders, o)
```

Comment:
At a glance, it looks like it would return more than `n` orders total, just `n` in each active/archived category.  That'll be unexpected for callers of `MarketOrders`, `AccountOrders`, and `Orders` where an `n` is specified.

If that assessment is correct, how about in `newestOrders` we sort the slice from `db.ordersView` and then truncate to `n`.

Reply:
I hadn't noticed this. Added a convoluted answer...